browser/ui/geolocation/geolocation_utils_mac.mm

```diff
@@ -0,0 +1,37 @@
+/* Copyright (c) 2024 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+#include "brave/browser/ui/geolocation/geolocation_utils.h"
+
+#import <CoreLocation/CoreLocation.h>
+
+namespace geolocation {
+
+bool IsSystemLocationSettingEnabled() {
+  // Service is off globally.
+  if (![CLLocationManager locationServicesEnabled]) {
+    return false;
+  }
+
+  CLLocationManager* manager = [[CLLocationManager alloc] init];
+  if (@available(macOS 11.0, *)) {
+    auto status = [manager authorizationStatus];
+    if (status == kCLAuthorizationStatusAuthorized) {
```

Comment:
I think it's good to link to the docs for this; I went to look at https://developer.apple.com/documentation/corelocation/clauthorizationstatus/kclauthorizationstatusauthorized

It shows as working on macOS 10.6 and newer (Lion I think); 11.0 would be Big Sur. Is that right?

Docs show `kCLAuthorizationStatusAuthorized` as deprecated for other platforms (macOS it's OK). What does the prompt look like for enable location services on macOS? Is it only yes/no? Or does macOS also support like iOS has `Always` or `When in use`?

Reply:
This code doesn't launch a prompt.  This code just check current state.
Maybe application launch that prompt. but not sure we want to do it?

Reply:
All addressed. PTAL again :)